recipes/libbasisu/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+
+required_conan_version = ">=1.33.0"
+
+
+class LibBasisUniversalConan(ConanFile):
+    name = "libbasisu"
+    description = "Basis Universal Supercompressed GPU Texture Codec"
+    homepage = "https://github.com/BinomialLLC/basis_universal"
+    topics = ("conan", "basis", "textures", "compression")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False],
+        "use_sse4": [True, False],
+        "with_zstd": [True, False],
+        "custom_iterator_debug_level": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+        "use_sse4": True,
+        "with_zstd": True,
+        "custom_iterator_debug_level": False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.compiler != "Visual Studio":
+            del self.options.custom_iterator_debug_level
+
+    def _minimum_compiler_version(self) -> bool:
+        return {
+            "Visual Studio": "15",
+            "gcc": "5.4",
+            "clang": "3.9",
+            "apple-clang": "10"
+        }
+ 
+    def validate(self):
+        min_version = self._minimum_compiler_version().get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        elif tools.Version(self.settings.compiler.version) < min_version:
+            raise ConanInvalidConfiguration("{} {} does not support compiler with version {} {}, minimum supported compiler version is {} ".format(self.name, self.version, self.settings.compiler, self.settings.compiler.version, min_version))
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["SSE4"] = self.options.use_sse4
+        self._cmake.definitions["ZSTD"] = self.options.with_zstd
+        custom_iterator_debug_level = self.options.get_safe("custom_iterator_debug_level", default="False")
+        self._cmake.definitions["BASISU_NO_ITERATOR_DEBUG_LEVEL"] = not custom_iterator_debug_level
```

Comment:
```suggestion
        self._cmake.definitions["BASISU_NO_ITERATOR_DEBUG_LEVEL"] = not self.options.get_safe("custom_iterator_debug_level", False)
```

Default won't be converted from String to Boolean.

Reply:
May I ask you for any possible solution for this? As I can see it's quite common in other recipes, e.g. `self._autotools.fpic = self.options.get_safe("fPIC", True)` in the [libpcap](https://github.com/conan-io/conan-center-index/blob/master/recipes/libcap/all/conanfile.py#L50) recipe and little bit lower in the [documentation](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#options) as well:
```
def build(self):
    # Will return None if doesn't exist
    fpic = self.options.get_safe("fPIC")
    # Will return the default value if the return is None
    shared = self.options.get_safe("shared", default=False)
```

Reply:
uilian suggested to change the argument of the `default` argument from the string `"False"` to the boolean `False`.
`.get_safe` tries to convert a value to the type of the default value.

Related to your question, I would do:
```python
fpic = self.options.get_safe("fPIC", True)  # Return True when fPIC is not available (it is often removed when built as a shared library)
```

Reply:
Correct. Thanks!